INSTALL

```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
 compilation:
 
     -Dwith-python=/bgsys/tools/Python-2.7/bin/hostpython
+or (much better)
+    -Dwith-python=/bgsys/local/python3/3.4.2/bin/python3
 
-Eventually, you also have to specify `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR`,
-if the they are not found, e.g.:
+CMake <3.4 is buggy about finding the matching libraries (for many years).
+Thus, you also have to specify `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR`
+if they are not found OR the incorrect libraries are found, e.g.:
 
     -DPYTHON_LIBRARY=/bgsys/tools/Python-2.7/lib64/libpython2.7.so.1.0
     -DPYTHON_INCLUDE_DIR=/bgsys/tools/Python-2.7/include/python2.7
+or (much better)
+    -DPYTHON_LIBRARY=/bgsys/local/python3/3.4.2/lib/libpython3.4m.a
+    -DPYTHON_INCLUDE_DIR=/bgsys/local/python3/3.4.2/include/python3.4m
+
+A complete `cmake` line for PyNEST could look like this:
+
+    module load gsl
+
+    cmake -DCMAKE_TOOLCHAIN_FILE=Platform/BlueGeneQ_XLC \
+          -DCMAKE_INSTALL_PREFIX=</install/path> \
+          -Dstatic-libraries=OFF \
+          -Dcythonize-pynest=OFF \
+	  -DCMAKE_C_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlc_r \
+	  -DCMAKE_CXX_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlcxx_r \
+	  -Dwith-python=/bgsys/local/python3/3.4.2/bin/python3 \
+	  -DPYTHON_LIBRARY=/bgsys/local/python3/3.4.2/lib/libpython3.4m.a \
+	  -DPYTHON_INCLUDE_DIR=/bgsys/local/python3/3.4.2/include/python3.4m \
+          -Dwith-ltdl=OFF \
+          <nest-src>
 
 Further, for running PyNEST, make sure all python dependencies are installed and
 environment variables are set properly:
 
+    module load python3/3.4.2
+    
     # adds PyNEST to the PYTHONPATH
     source <nest-install-dir>/bin/nest_vars.sh
+    
     # makes HOME and PYTHONPATH available for python
-    runjob --exp-env HOME --exp-env PYTHONPATH ... : python script.py
+    runjob \
+    --exp-env HOME \
+    --exp-env PATH \
+    --exp-env LD_LIBRARY_PATH \
+    --exp-env PYTHONUNBUFFERED \
+    --exp-env PYTHONPATH \
+    ... \
+    : /bgsys/local/python3/3.4.2/bin/python3.4 script.py
+
+BlueGene/Q and GCC
+------------------
+
+Compiling NEST with GCC (`-DCMAKE_TOOLCHAIN_FILE=Platform/BlueGeneQ_GCC`)
+might require you to use a GSL library compiled using GCC, otherwise undefined
+symbols break your build. After the GSL is build with GCC and installed in
+<gsl-install-dir>, add `-Dwith-gsl=<gsl-install-dir>` to the `cmake` line.
 
 Compiling for Fujitsu Sparc64
```

Comment:
Please also put the code blocks in this section in triple backticks and remove their indentation